src/uniprotkb/components/protein-data-views/FreeTextView.tsx

```diff
@@ -20,50 +21,78 @@ const needsNewLineRE = /^\).\s+/;
 
 const getEntryPathForCitation = getEntryPathFor(Namespace.citations);
 
+export const ACCommentView = ({ string }: { string: string }) => {
+  // This replaces any occurrences of "AC <accession>" with "AC "<link to the accession>
+  const tokens = string.split(' ');
+  const nodes: ReactNode[] = new Array(tokens.length);
+  for (let i = 0; i < tokens.length; i += 1) {
+    const accession = tokens[i].match(uniProtKBAccessionRegEx)?.[0];
+    nodes[i] =
+      i > 0 && tokens[i - 1] === 'AC' && accession ? (
+        <Link key={i} to={getEntryPath(Namespace.uniprotkb, accession)}>
+          {accession}
+        </Link>
+      ) : (
+        tokens[i]
+      );
+  }
+  const lastIndex = tokens.length - 1;
+  if (!tokens[lastIndex].endsWith('.')) {
+    nodes[lastIndex] = <Fragment key="last">{nodes[lastIndex]}.</Fragment>;
+  }
+  return <>{nodes.reduce((prev, curr) => [prev, ' ', curr])}</>;
+};
+
 type TextViewProps = { comments: TextWithEvidence[]; noEvidence?: boolean };
 
-export const TextView = ({ comments, noEvidence }: TextViewProps) => (
-  <div className="text-block">
-    {comments.map((comment, index) => (
-      // eslint-disable-next-line react/no-array-index-key
-      <Fragment key={index}>
-        {comment.value.split(pubMedRE).map((part, index, { length }) => {
-          /** We should get odds plain text, and evens pubmed ID, but we are
-           *  still double-checking just in case */
-          if (pubMedIDRE.test(part)) {
-            // PubMed ID, insert a link
-            return (
-              // eslint-disable-next-line react/no-array-index-key
-              <Link key={index} to={getEntryPathForCitation(part)}>
-                {part}
-              </Link>
-            );
-          }
-          if (needsNewLineRE.test(part)) {
-            // add new line before adding the rest of the plain text
-            return (
-              // eslint-disable-next-line react/no-array-index-key
-              <Fragment key={index}>
-                ).
-                <br />
-                {part.replace(needsNewLineRE, '')}
-              </Fragment>
-            );
-          }
-          // If the last section doesn't end with a period, add it
-          if (index + 1 === length && !part.endsWith('.')) {
-            return `${part}.`;
-          }
-          // use plain text as such
-          return part;
-        })}
-        {!noEvidence && comment.evidences && (
-          <UniProtKBEvidenceTag evidences={comment.evidences} />
-        )}
-      </Fragment>
-    ))}
-  </div>
-);
+export const TextView = ({ comments, noEvidence }: TextViewProps) => {
+  const acRE = new RegExp(`AC ${uniProtKBAccessionRegEx.source}`, 'gi');
+  return (
+    <div className="text-block">
+      {comments.map((comment, index) => (
+        // eslint-disable-next-line react/no-array-index-key
+        <Fragment key={index}>
+          {comment.value.split(pubMedRE).map((part, index, { length }) => {
```

Comment:
Thinking it might be cleaner to use the exact same strategy used for PubMED ids: split the comment on acRE and looking at exactly matching groups?

Reply:
Will have a go at refactoring

Reply:
Done

Reply:
Thanks, that's nicer.
The other thing I was referring to (but maybe it's not an issue) is the fact the comments are split on `pubMedRE` (L59) and then evaluated for `aceRE` (L71). I was wondering if it would be better to do another split on the comment values (as another block) afterward? It could be more readable, but maybe less performant?

Reply:
I see your point and that was also something in the back of my head. After the `pubMedRE` we'll have an array of react elements and strings which I anticipate will complicate it a bit. For each element of the array we could check the type and if it's a string go ahead and split on `acRE`. What do you think?

Reply:
I'm not sure there is an easy way to check for `string|ReactNode`. If it gets too complicated don't bother.

Reply:
I think this is good to go now